modules/plugin/jdbc/jdbc-spatialite/src/test/java/org/geotools/data/spatialite/SpatiaLiteDataStoreAPITestSetup.java

```diff
@@ -28,7 +30,16 @@
     protected SpatiaLiteDataStoreAPITestSetup() {
         super(new SpatiaLiteTestSetup());
     }
-    
+
+    protected void setUpData() throws Exception {
+        //kill all the data
+        super.setUpData();
+        try {
+            dropDataTypesTable();
```

Comment:
Agreed, conceptually.

But just followed what I found on its superclass (and other related classes):
https://github.com/geotools/geotools/blob/master/modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCDataStoreAPITestSetup.java#L38

This setUpData method is different from setUp method, both called by "connect" method in JDBCTestSupport, which in turn is called in the setUp method from OnlineTestCase. This last one is the true junit setup method. There are equivalent tearDown/disconnect methods all the way down, but rarely implemented. Just guessing, but given the nontrivial construct, there may be cases (uncaught exceptions during setup?) where the teardown code is not reached.

I can try to change it, but I'd rather be conservative and stick to what is already there.